semeio/jobs/design_kw/design_kw.py

```diff
@@ -79,15 +79,50 @@ def is_comment(line):
 
 
 def extract_key_value(parameters):
-    # Extract all key, value pairs from parameters file
-    # If key already exists, raise Exception
+    """Parses a list of strings, looking for key-value pairs pr. line
+    separated by whitespace, into a dictionary.
 
+    Spaces in either key or in value are not supported.
+
+    Args:
+        parameters (list of str)
+
+    Returns:
+        dict, with the keys and values parsed.
+
+    Raises:
+        ValueError, with error messages and all unparseable lines.
+    """
     res = {}
+    unparsable_lines = []
+    errors = []
     for line in parameters:
-        key, value = line.split()
+        try:
+            key, value = line.split()
+        except ValueError as e_msg:
+            if "not enough values to unpack" in str(
+                e_msg
+            ) or "need more than 1 value to unpack" in str(e_msg):
+                # (py2 and py3 differ in emitted error message)
+                errors += ["No value found in line {}".format(line)]
+            elif "too many values to unpack" in str(e_msg):
+                errors += ["Spaces in values are not supported"]
+            else:
+                errors += [str(e_msg)]
+            unparsable_lines += [line]
+            continue
         if key in res:
-            raise SystemExit(
-                "{} is defined multiple times in parameters file".format(key)
-            )
+            errors += ["{} is defined multiple times".format(key)]
+            continue
         res[key] = value
+    if unparsable_lines or errors:
+        error_str = ""
+        if unparsable_lines:
+            error_str = "Could not parse these lines:\n"
+            error_str += "\n".join(unparsable_lines)
+        if '"' in error_str:
```

Comment:
Do you think this kind of error is the most likely to pop up as you are special casing for it? If you think it is likely that it will happen again I´m fine with special casing it. But if the error from above is enough to clue to user in, it might not be needed?

Reply:
I  anticipate that users might try to add quotes if they need strings with spaces as values in  `parameters.txt`, so as long as it is not supported, we can give an (overly?) helpful message.